magda-io/magda
# Issue/1717 direct non prod emails to default mailbox

### What this PR does

Fixes #1717 

Made preview and dev ALWAYS send to the default recipient even if a contactPoint is available, but put the normal recipient in the subject line so you can still test that it _would've_ gone to the right place.

### TO TEST
1. Check out the branch
2. Change `correspondence-api` on the preview server to always send to default recipient

```
helm upgrade issue-1717-direct-non-prod-emails-to-default-mailbox deploy/helm/magda --install --no-hooks --namespace issue-1717-direct-non-prod-emails-to-default-mailbox -f deploy/helm/preview.yml --set global.image.repository=registry.gitlab.com/magda-data/magda/data61,global.image.tag=issue-1717-direct-non-prod-emails-to-default-mailbox,ingress.hostname=issue-1717-direct-non-prod-emails-to-default-mailbox.dev.magda.io,correspondence-api.alwaysSendToDefaultRecipient=true --timeout 3600 --wait
```

3. Go to https://issue-1717-direct-non-prod-emails-to-default-mailbox.dev.magda.io/dataset/ds-dga-419aa51f-5c8b-42a0-b76d-5a9f1c2ae1d4/details?q=. This has the contactPoint altered to a mailinator address. Ask a question about this dataset

4. Check that nothing goes to the mailinator inbox: https://www.mailinator.com/v2/inbox.jsp?zone=public&query=magda-test

5. Check that the email did go to the default recipient, which is the google group: https://groups.google.com/forum/#!forum/magda-test

6. Check that the subject line has "For magda-test@mailinator.com" in it.

7. Make the config NOT always send to default:

```
helm upgrade issue-1717-direct-non-prod-emails-to-default-mailbox deploy/helm/magda --install --no-hooks --namespace issue-1717-direct-non-prod-emails-to-default-mailbox -f deploy/helm/preview.yml --set global.image.repository=registry.gitlab.com/magda-data/magda/data61,global.image.tag=issue-1717-direct-non-prod-emails-to-default-mailbox,ingress.hostname=issue-1717-direct-non-prod-emails-to-default-mailbox.dev.magda.io,correspondence-api.alwaysSendToDefaultRecipient=false --timeout 3600 --wait
```

8. Repeat the previous steps, but this time the email should go to the mailinator address and _not_ to the google group

### Checklist
